requirements.txt

```diff
@@ -23,5 +23,8 @@ pytest~=8.3.4
 pytest-cov~=6.0.0
 pytest-xdist~=3.6.1
 pyyaml~=6.0.2
-semgrep==1.104.0
+# If Python == 3.9, install Semgrep 1.52.0
+semgrep==1.52.0; python_version == "3.9"
```

Comment:
Not sure I'm ok with this. It's 50 versions out of date. Do we know if it has any known vulnerabilities or issues?

Reply:
I don't know anything about it, but to support Python 3.9, there's a conflict with boto2, semgrep, and the version of urllib3 they use:
    semgrep 1.107.0 depends on urllib3~=2.0
    botocore 1.36.18 depends on urllib3<1.27 and >=1.25.4; python_version < "3.10"
    
  Removing the version restrictions from requirments.txt and then pip resolves with semgrep 1.52 and botocore 1.36.22. I tried pinning semgrep to latest and letting pip sort out boto3, but that didn't work.
  
  We could pin semgrep and just not install boto3 for python3.9?